agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java

```diff
@@ -202,13 +203,17 @@ public TelemetryItem map(SpanData span, float samplingPercentage) {
   private TelemetryItem exportRemoteDependency(
       SpanData span, boolean inProc, float samplingPercentage) {
     RemoteDependencyTelemetryBuilder telemetryBuilder = RemoteDependencyTelemetryBuilder.create();
-    telemetryInitializer.accept(telemetryBuilder);
 
     // set standard properties
     setOperationTags(telemetryBuilder, span);
     setTime(telemetryBuilder, span.getStartEpochNanos());
     setSampleRate(telemetryBuilder, samplingPercentage);
+
+    // update tags
+    ResourceParser.updateRoleNameAndInstance(telemetryBuilder, span.getResource());
+    telemetryInitializer.accept(telemetryBuilder);
```

Comment:
I think it's clearer to leave initialization of the telemetry builder at the top, before updating the telemetry builder

Reply:
i can move line 214 to the top, but inside updateRoleNameAndInstance, i can override the value if it has been set. not sure builder let me retrieve it though.

Reply:
or move both to the top and keep the order?

Reply:
i moved it back to the top and add a check in the builder.. only override it if it's not set yet. i think it's cleaner this way so that it is not depended on the order of code execution.

Reply:
i am going to add a test for that.